invokeai/backend/model_manager/load/model_cache/model_cache_base.py

```diff
@@ -46,6 +46,8 @@ class CacheRecord(Generic[T]):
 
     key: str
     model: T
+    device: torch.device
+    state_dict: Optional[Dict[str, torch.Tensor]]  # this is a copy that stays in CPU
```

Comment:
I've added documentation for how the process works. It might be a good idea to declare `state_dict` a read-only attribute, since it should never be modified directly. Is there a good way to do this with a dataclass? I see a number of ugly-looking recipes on the web.